tests/db/model/generatortemplate/classMethods.js

```diff
@@ -0,0 +1,202 @@
+      gt.name += '1';
+      GeneratorTemplate.create(gt)
+      .then((o) => {
+        createdGT = o;
+        return GeneratorTemplate.getSemverMatch(gt.name, '1.0.0');
+      })
+      .then((o) => {
+        if (!o) {
+          return done('Expection a GeneratorTemplate object');
+        }
+
+        expect(o.dataValues.version).to.equal(createdGT.dataValues.version);
+        return done();
+      })
+      .catch(done);
+    });
+
+    it('match version when GT version = 2.0.0 and test ' +
+      'verison = ^2.0.0', (done) => {
+      let createdGT;
+      gt.version = '2.0.0';
+      gt.name += '2';
+      GeneratorTemplate.create(gt)
+      .then((o) => {
+        createdGT = o;
+        return GeneratorTemplate.getSemverMatch(gt.name, '^2.0.0');
+      })
+      .then((o) => {
+        if (!o) {
+          return done('Expection a GeneratorTemplate object');
+        }
+
+        expect(o.dataValues.version).to.equal(createdGT.dataValues.version);
+        return done();
+      })
+      .catch(done);
+    });
+
+    it('match version when GT version = 3.0.0 and test ' +
+      'verison is <=3.0.0', (done) => {
+      let createdGT;
+      gt.version = '3.0.0';
+      gt.name += '3';
+      GeneratorTemplate.create(gt)
+      .then((o) => {
+        createdGT = o;
+        return GeneratorTemplate.getSemverMatch(gt.name, '<=3.0.0');
+      })
+      .then((o) => {
+        if (!o) {
+          return done('Expection a GeneratorTemplate object');
+        }
+
+        expect(o.dataValues.version).to.equal(createdGT.dataValues.version);
+        return done();
+      })
+      .catch(done);
+    });
+
+    it('no match when GT version = 5.0.0 and test ' +
+      'verison is <=4.0.0', (done) => {
+      gt.version = '5.0.0';
+      gt.name += '5';
+      GeneratorTemplate.create(gt)
+      .then(() => GeneratorTemplate
+        .getSemverMatch(gt.name, '<=4.0.0'))
+      .then((o) => {
+        if (!o) {
+          return done();
+        }
+
+        return done('Expecting no sample generatorTemplate to be returned');
+      })
+      .catch(done);
+    });
+
+    it('no match when GT version = 4.0.0 and test ' +
+      'verison is >4.0.0', (done) => {
+      gt.version = '4.0.0';
+      gt.name += '4';
+      GeneratorTemplate.create(gt)
+      .then(() => GeneratorTemplate.getSemverMatch(gt.name, '>4.0.0'))
+      .then((o) => {
+        if (!o) {
+          return done();
+        }
+
+        return done('Expecting no sample generatorTemplate to be returned');
+      })
+      .catch(done);
+    });
+
+    it('no match when GT matching the name is not found', (done) => {
+      const gtName = 'removeRandomSGTName';
+      GeneratorTemplate.getSemverMatch(gtName, '1.0.0')
+      .then((o) => {
+        if (!o) {
+          return done();
+        }
+
+        return done('Expecting no sample generatorTemplate to be returned');
+      })
+      .catch(done);
+    });
+
+    it('no match when GT version = 5.0.0 and test ' +
```

Comment:
Is this test repeated?

Reply:
Yes, this is a duplicate. Removed this test in the latest commit.